packages/channel-provider/src/messaging-service.ts

```diff
@@ -103,7 +103,7 @@ export class MessagingService {
         window.removeEventListener('message', listener);
         logger.info({response: event.data}, 'Received response');
         resolve(event.data.result);
-      } else if (isJsonRpcErrorResponse(event.data)) {
+      } else if (event.data && event.data.jsonrpc && isJsonRpcErrorResponse(event.data)) {
```

Comment:
Should this runtime check happen in `isJsonRpcErrorResponse`

Reply:
Another proposal for this issue: https://github.com/statechannels/monorepo/pull/1994